### PR TITLE
Add documentation for StatsD component

### PR DESCRIPTION
**Pull request by Links2004**
## Description:

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#6642

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by Links2004** [comment]
Waiting for review and merge, not stale

File: components/statsd.rst [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    :description: Instructions for setting up a StatsD
+    :keywords: StatsD, metrics
+
+StatsD is a `protocol <https://github.com/statsd/statsd/blob/master/docs/metric_types.md>`_ to send metrics to a Daemon to store and aggregate them.
+Today there are many monitoring solutions that support receiving metrics via the StatsD protocol.
+
+.. note::
+
+    This component will only work with Arduino for the ESP8266 since the `esphome::socket` implementation does not support UDP.
+    ESP32 will work fine with or without Arduino as Framework.
```
Comment (jesserockz) [inline components/statsd.rst]: Not required as there is no alternatives for users to use =)
```suggestion
```

File: components/statsd.rst [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      prefix: esphome
+      sensors:
+        id: wlan_signal
+        name: test1.wifi.signal
+
+    sensor:
+      platform: wifi_signal
+      id: wlan_signal
+      name: "WiFi Signal"
+      update_interval: 10s
```
Comment (jesserockz) [inline components/statsd.rst]: It is preferred that these examples be sparse as they should only show basic/required syntax and not optional fields.

If full examples are desired, they can go at the bottom of the page under the configuration variables.

```suggestion
    statsd:
      host: REPLACEME
      sensors:
        id: some_sensor
        name: test1.sensor

    sensor:
      platform: ...
      id: some_sensor
```

File: components/statsd.rst [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+
+This example will generate a metric named `esphome.test1.wifi.signal` with the value of the `wifi_signal` sensor.
```
Comment (jesserockz) [inline components/statsd.rst]:
```suggestion
This example will generate a metric named `test1.sensor` with the value of the `some_sensor` sensor.
```

File: components/statsd.rst [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+- **update_interval** (*Optional*, uint16): How often to send the metrics. Defaults to ``10s``.
+- **sensor** (*Optional*, :ref:`sensors`): A list of sensors to generate metrics for.
+- **binary_sensor** (*Optional*, :ref:`sensors`): A list of binary sensors to generate metrics for.
+
+.. _sensors:
+
+Sensor list
+-----------
+
+- **id** (**Required**, id): The ID of the sensor.
```
Comment (jesserockz) [inline components/statsd.rst]:
```suggestion
- **id** (**Required**, :ref:`config-id`): The ID of the sensor.
```

File: components/statsd.rst
Comment (jesserockz) [inline components/statsd.rst:1]: This page is missing the `see also` section at the bottom with at a minimum: 
```rst
See Also
--------

- :ghedit:`Edit`
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: components/statsd.rst
Comment (Links2004) [reply components/statsd.rst:1]: added

**Review by jesserockz — APPROVED** [review]

**Pull request merged**